.gitlab-ci.yml

```diff
@@ -0,0 +1,74 @@
+stages:
+  - build
+  - test
+  - report
+
+before_script:
+  - sh -c 'echo "deb [arch=amd64,arm64] http://repo.ros2.org/ubuntu/main `lsb_release -cs` main" > /etc/apt/sources.list.d/ros2-latest.list'
+  - curl http://repo.ros2.org/repos.key | apt-key add -
+  - apt-get update
+  - apt-get install -y python3-colcon-common-extensions
+  - source /opt/ros/crystal/setup.bash
+
+build_launchtest:
+  stage: build
+  image: osrf/ros2:nightly
+  script:
+    - colcon build --packages-up-to apex_launchtest apex_launchtest_cmake
+  artifacts:
+    paths:
+      - .
+
+build_all:
+  stage: build
+  image: osrf/ros2:nightly
+  script:
+    - colcon build
+  artifacts:
+    paths:
+      - .
+
+test_launchtest:
+  stage: test
+  image: osrf/ros2:nightly
+  dependencies:
+    - build_launchtest
+  script:
+    - pip3 install mock
+    - colcon test --packages-select apex_launchtest apex_launchtest_cmake
+    - colcon test-result --verbose
+  artifacts:
+    when: always
+    paths:
+      - log
+
+test_all:
+  stage: test
+  image: osrf/ros2:nightly
+  dependencies:
+    - build_all
+  script:
+    - pip3 install mock
+    - colcon test --pytest-args "--cov"
+    - colcon test-result --verbose
+  artifacts:
+    when: always
+    paths:
+      - .
```

Comment:
Same here. Maybe only upload the part that we care, i.e., log?

Reply:
I can't change this one - the .coverage files are generated in the package roots and I need those to be passed along.  Is there a good way to tar up the .coverage files and then untar them back into the same directories without having to explicitly list apex_launchtest, apex_launchtest_cmake, and apex_launchtest_ros?  I'm concerned that in the future if another package is added, it will get missed for coverage

Reply:
You can use `--cov-report` to specify the output directory. More info can be found here https://pytest-cov.readthedocs.io/en/latest/reporting.html

That being said, it's up to you to decide whether or not to use it.

Reply:
re: the last part - I don't want to generate coverage unless everything passes.